frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx

```diff
@@ -0,0 +1,179 @@
+    const provisioner: string = updatedStorageClass?.provisioner || '';
+    // if the provisioner is unknown or no storage class selected, user should be able to set any access mode
+    const modes = provisionerAccessModeMapping[provisioner]
+      ? provisionerAccessModeMapping[provisioner]
+      : getAccessModeForProvisioner(provisioner);
+    // setting message to display for various modes when a storage class of a know provisioner is selected
+    const displayMessage =
+      provisionerAccessModeMapping[provisioner] || isCephProvisioner(provisioner)
+        ? 'Access mode is set by storage class and cannot be changed'
+        : 'Permissions to the mounted drive';
+    setAccessMode('ReadWriteOnce');
+    setAccessModeHelp(displayMessage);
+    // setting accessMode to default with the change to Storage Class selection
+    setAllowedAccessModes(modes);
+    setStorageClass(updatedStorageClass?.metadata?.name);
+    setStorageProvisioner(provisioner);
+    if (storageProvisioner.includes(cephStorageProvisioners[1])) {
+      setVolumeMode('Filesystem');
+    }
+  };
+
+  React.useEffect(() => {
+    const volumeClaimTemplate = {
+      spec: {
+        accessModes: [accessMode],
+        storageClassName: storageClass,
+        volumeMode,
+        resources: {
+          requests: {
+            storage: `${requestSizeValue}${requestSizeUnit}`,
+          },
+        },
+      },
+    };
+    setFieldTouched(name);
+    setFieldValue(name, volumeClaimTemplate);
+  }, [
+    volumeMode,
+    accessMode,
+    requestSizeValue,
+    requestSizeUnit,
+    storageClass,
+    name,
+    setFieldTouched,
+    setFieldValue,
+  ]);
+
+  return (
+    <ExpandCollapse
+      textExpanded={t('pipelines-plugin~Hide VolumeClaimTemplate options')}
+      textCollapsed={t('pipelines-plugin~Show VolumeClaimTemplate options')}
+    >
+      <StorageClassDropdown
+        onChange={handleStorageClass}
+        id="storageclass-dropdown"
+        data-test="storageclass-dropdown"
+        describedBy="storageclass-dropdown-help"
+        required={false}
+        valueFromAST
+        name="storageClass"
+      />
+      <Label className="control-label co-required" htmlFor="access-mode">
+        {t('pipelines-plugin~Access Mode')}
+      </Label>
+      <FormGroup fieldId="accessMode">
+        {accessModeRadios.map((radio) => {
+          let radioObj = null;
+          const disabled = !allowedAccessModes.includes(radio.value);
+
+          allowedAccessModes.forEach((mode) => {
+            const checked = !disabled ? radio.value === accessMode : radio.value === mode;
+            radioObj = (
+              <RadioInput
+                {...radio}
+                key={radio.value}
+                onChange={handleAccessMode}
+                inline
+                disabled={disabled}
+                checked={checked}
+                aria-describedby="access-mode-help"
+                name="accessMode"
+              />
+            );
```

Comment:
Doing a `.forEach` in the middle of rendering is mighty weird... are you not just going to override your rendering multiple times before rendering?

Reply:
Moved the forEach out and only applying to set the `checked` flag. I am replicating the same logic used in createPVCForm.